service/controller/resource/ensurecpcrs/create.go

```diff
@@ -0,0 +1,72 @@
+package ensurecpcrs
+
+import (
+	"context"
+	"fmt"
+
+	infrastructurev1alpha2 "github.com/giantswarm/apiextensions/pkg/apis/infrastructure/v1alpha2"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/aws-operator/service/controller/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCluster(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensuring %#q CR exists", fmt.Sprintf("%T", infrastructurev1alpha2.AWSControlPlane{})))
+
+		exists, err := r.awsControlPlaneCRExists(ctx, cr)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		if !exists {
+			err := r.createAWSControlPlaneCR(ctx, cr)
```

Comment:
General question:
Why does `aws-operator` create this CR? Is this an intermediate solution?
I assume that `aws-operator` also consumes this `CR` - so this currently looks smelly to me.

Maybe I missed some discussion here.

Reply:
Thanks for asking. This may help: https://github.com/giantswarm/aws-operator/pull/2157.